odoo/service/model.py

```diff
@@ -35,15 +35,38 @@ def __str__(self):
         return ', '.join(params)
 
 
-def call_kw(model: BaseModel, name: str, args: list, kwargs: Mapping):
-    """ Invoke the given method ``name`` on the recordset ``model``. """
+def public_model_method(model: BaseModel, name: str):
+    """ Get a public method from a model.
+
+    When the method does not exist or is inaccessible, raise appropriate errors.
+    Accessible methods are public (in sense that python defined it:
+    not prefixed with "_") and are not decorated with `@api.private`.
+    """
+    assert isinstance(model, BaseModel)
     method = getattr(model, name, None)
-    if not method:
-        raise AttributeError(f"The method '{name}' does not exist on the model '{model._name}'")
+    if not callable(method):
+        raise AttributeError(model.env._("The method '%s' does not exist", f'{model._name}.{name}'))  # noqa: TRY004
+
+    for cla in type(model).mro():
+        if not (cla_method := getattr(cla, name, None)):
+            break
+        api = getattr(cla_method, '_api', '')
+        if 'private' in api or name.startswith('_'):
+            raise AccessError(model.env._("Private methods (such as '%s') cannot be called remotely.", f"{model._name}.{name}"))
```

Comment:
Same, doesn't it make sense to translate this error since it is a Programming error ?

Reply:
Lint forces to translate it.